#!/usr/bin/env python
PACKAGE = "youbot_kinematics_rviz"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

#       Name    Type    Level 	Description           Default    Min   Max
gen.add("xd", double_t, 0, "desired x coordinate", 0.4, -0.627, 0.627)
gen.add("yd", double_t, 0, "desired y coordinate", 0, -0.627, 0.627)
gen.add("zd", double_t, 0, "desired z coordinate", 0.3, -0.2918, 0.7412)
gen.add("pitchd", double_t, 0, "desired pitch angle in degree", 0, -180, 180)
gen.add("rolld", double_t, 0, "desired roll abgle in degree", 0, -165, 165)

gen.add("arm_to_front", bool_t, 0, "arm face to front", True)
gen.add("arm_bended_up", bool_t, 0, "arm is bended up", True)


exit(gen.generate(PACKAGE, "youbot_kinematics_rviz", "work_space_state"))
